pysindy/feature_library/polynomial_library.py

```diff
@@ -222,66 +226,81 @@ def transform(self, x):
         """
         check_is_fitted(self)
 
-        x = check_array(x, order="F", dtype=FLOAT_DTYPES, accept_sparse=("csr", "csc"))
+        xp_full = []
+        for x in x_full:
+            if sparse.issparse(x) and x.format not in ["csr", "csc"]:
```

Comment:
The `PolynomialLibrary` is the only library that supports the sparse input data currently. I don't understand the use case for it--what sort of time series data is expected to be sparse? If there is a usage, should we make an issue to implement it more generally?

Reply:
The `PolynomialLibrary` is the only library that _optimizes_ for sparse input.  Sparse input may also work in other libraries via duck typing.

I think Kathleen added these lines when I ran git blame.  I had the same question.  It was just easier for me to build backwards compatibility rather than try to find out the reason.

I'll create an issue and reach out to Kathleen to see why she added them.

Reply:
Cool, makes sense! Obviously, nothing required for this PR, but just an observation.